TypeCal.jacc

```diff
@@ -66,6 +71,10 @@ assign: ref EQUAL expr
 {
     $$ = new AssignmentNode((ReferenceNode)$1, (ParseNode)$3);
 }
+| declaration EQUAL expr
+{
+    $$ = new AssignmentNode((ReferenceNode)((DeclarationNode)$1).evaluate(), (ParseNode)$3);
+}
```

Comment:
Add a case to the assignment node to account for a declaration on the left side, not a record declaration though.
DeclarationNode evaluates to a ReferenceNode representing the variable that was just declared, in the global scope though.